lale/lib/autogen/factor_analysis.py

```diff
@@ -38,17 +38,24 @@ def transform(self, X):
                 "random_state",
             ],
             "relevantToOptimizer": [
-                "n_components",
                 "tol",
                 "copy",
-                "max_iter",
+                "rotation",
                 "svd_method",
                 "iterated_power",
             ],
             "additionalProperties": False,
             "properties": {
                 "n_components": {
-                    "enum": ["int", None],
+                    "anyOf": [
+                        {
+                            "type": "integer",
+                            "minimumForOptimizer": 2,
```

Comment:
There should probably also be a `minimum`, not just a `minimumForOptimizer`. Need to figure out what that minimum is, based on dynamic exceptions.